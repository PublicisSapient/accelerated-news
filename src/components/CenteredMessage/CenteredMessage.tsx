import React from 'react';
import { CenteredContainer } from '../Containers';

export const CenteredMessage: React.FC = ({ children }) => {
  return (
    <CenteredContainer className="p-3">
      <h1 className="title">{children}</h1>
    </CenteredContainer>
  );
};
