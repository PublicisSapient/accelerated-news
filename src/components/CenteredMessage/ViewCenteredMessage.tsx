import React from 'react';
import { ViewCenteredContainer } from '../Containers';

export const ViewCenteredMessage: React.FC = ({ children }) => {
  return (
    <ViewCenteredContainer className="p-3">
      <h1 className="title">{children}</h1>
    </ViewCenteredContainer>
  );
};
