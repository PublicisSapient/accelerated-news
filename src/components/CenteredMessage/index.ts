export * from './CenteredMessage';
export * from './ViewCenteredMessage';
