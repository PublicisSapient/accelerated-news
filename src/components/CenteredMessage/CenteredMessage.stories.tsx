import React from 'react';
import { Meta } from '@storybook/react';
import { ViewVerticalContainer } from '../Containers';
import { SimpleHeader } from '../Header';
import { CenteredMessage } from './CenteredMessage';
import { ViewCenteredMessage } from './ViewCenteredMessage';

export default {
  title: 'Components/CenteredMessage',
  component: CenteredMessage,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const CenteredMessageStory = () => (
  <ViewVerticalContainer>
    <SimpleHeader />
    <CenteredMessage>Hello!</CenteredMessage>
  </ViewVerticalContainer>
);
CenteredMessageStory.storyName = 'CenteredMessage';

export const ViewCenteredMessageStory = () => (
  <ViewCenteredMessage>Hello!</ViewCenteredMessage>
);
ViewCenteredMessageStory.storyName = 'ViewCenteredMessage';
