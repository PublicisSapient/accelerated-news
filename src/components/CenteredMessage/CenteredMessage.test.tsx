import React from 'react';
import { CenteredMessage } from './CenteredMessage';
import { ViewCenteredMessage } from './ViewCenteredMessage';
import { render } from '@testing-library/react';

describe('CenteredMessage', () => {
  test('CenteredMessage renders correctly', () => {
    const { asFragment } = render(
      <CenteredMessage>Hello World!</CenteredMessage>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('ViewCenteredMessage renders correctly', () => {
    const { asFragment } = render(
      <ViewCenteredMessage>Hello World!</ViewCenteredMessage>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
