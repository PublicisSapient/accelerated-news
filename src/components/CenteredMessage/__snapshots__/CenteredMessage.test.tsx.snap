// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenteredMessage CenteredMessage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex justify-center items-center text-center p-3"
    data-testid="centered-container"
  >
    <h1
      class="title"
    >
      Hello World!
    </h1>
  </div>
</DocumentFragment>
`;

exports[`CenteredMessage ViewCenteredMessage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="h-screen flex justify-center items-center text-center p-3"
    data-testid="centered-container"
  >
    <h1
      class="title"
    >
      Hello World!
    </h1>
  </div>
</DocumentFragment>
`;
